### PR TITLE
Events collection page (except filters)

### PR DESCRIPTION
**Pull request by tomkingston**
## Description of change

Sets up the events collection list page in react - including:

- Render each item in the list with badges and metadata
- Sort by options (note that the default sort by will be applied when the filters are added)
- Add event button

## Test instructions

Go to `events/react`

The collection list should appear, with sort by options and a button to add an event.

## Screenshots

Screenshot from 2021-06-21 17-07-08

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/apps/events/client/transformers.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  start_date,
+  end_date,
+  organiser,
+  lead_team,
+  uk_region,
+  disabled_on,
+} = {}) => {
+  if (!id) {
+    return
+  }
```
Comment (paulgain) [inline src/apps/events/client/transformers.js:22]: When does an event not have an `id`? We should do the same for what we did for contacts and run a quick db query. We don't want to be copying over bad habits.

**Review by paulgain — APPROVED** [review]
LGTM

File: src/apps/events/client/transformers.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      label: 'Lead team',
+      value: get(lead_team, 'name'),
+    })
+  }
+  if (organiser) {
+    metadata.push({
+      label: 'Organiser',
+      value: get(organiser, 'name'),
+    })
+  }
```
Comment (ian-leggett) [inline src/apps/events/client/transformers.js:53]: Does feel like this could go into a function and ideally we could avoid the mutation.

**Review by ian-leggett — APPROVED** [review]

File: test/functional/cypress/specs/events/sort-spec-react.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    })
+
+    it('should sort by "Earliest start date"', () => {
+      cy.get(element).select('start_date:asc')
+      cy.wait('@apiRequest').then(({ request }) => {
+        expect(request.body.sortby).to.equal('start_date:asc')
+      })
+    })
+
+    it('should sort by "Least recently updated"', () => {
```
Comment (elcct) [inline test/functional/cypress/specs/events/sort-spec-react.js:69]: This should read "should sort by `Latest start date`"?

**Pull request merged**